import_precheck/check_osversion.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"fmt"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/crunk1/golang-sak"
+)
+
+var OSVERSIONS = []string{
+	"Windows Server 2008 R2",
+	"Windows Server 2012 R2",
+	"Windows Server 2016 R2",
+	"Debian GNU/Linux 8",
+	"Debian GNU/Linux 9",
+	"Red Hat Enterprise Linux Server release 6",
+	"Red Hat Enterprise Linux Server release 7",
+	"CentOS release 6",
+	"CentOS Linux release 7",
+}
+
+type OSVersionCheck struct {}
+
+func (c *OSVersionCheck) GetName() string {
+	return "OS Version Check"
+}
+
+func (c *OSVersionCheck) Run() (*Report, error) {
+	r := &Report{Name: c.GetName()}
+	v, err := c.getOSVersion()
+	if err != nil {
+		return nil, err
+	}
+	found := sak.Any(OSVERSIONS, func(i interface{}) interface{} {
+		if strings.Contains(v, i.(string)) {
+			return v
+		}
+		return nil
+	})
+	if found == nil {
+		msg := fmt.Sprintf("OS Version %q is not supported.", v)
+		r.Incompatibilities = append(r.Incompatibilities, msg)
+	}
+	return r, nil
+}
+
+func (c *OSVersionCheck) getOSVersion() (string, error) {
+	goos := runtime.GOOS
+	if goos == "windows" {
+		return c.getWindowsInfo()
+	}
+
+	out, err := exec.Command("cat", "/etc/redhat-release").Output()
+	if err == nil {
+		return string(out), err
+	}
+
+	out, err = exec.Command("cat", "/etc/issue").Output()
```

Comment:
Look at /etc/os-release instead

Reply:
Actually this is more complicated than that. Remind me on Monday to share the code I have for the inventory agent.

Reply:
Also ioutil.readfile